src/conditions/base/time.ts

```diff
@@ -1,23 +1,15 @@
-import Joi from 'joi';
+import { z } from 'zod';
 
-import { omit } from '../../utils';
+import { rpcConditionSchema } from './rpc';
 
-import { RpcCondition, rpcConditionRecord } from './rpc';
+// TimeCondition is an RpcCondition with the method set to 'blocktime' and no parameters
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const { parameters: _, ...restShape } = rpcConditionSchema.shape;
```

Comment:
Interesting; so this is how you remove parameters from being a schema entry using zod.